example-library/src/main/java/uk/gov/gchq/palisade/example/library/common/ExampleUser.java

```diff
@@ -76,16 +73,16 @@ public ExampleUser trainingCompleted(final EnumSet<TrainingCourse> trainingCompl
      *
      * @return the {@link EnumSet} of training courses
      */
-    public EnumSet<TrainingCourse> getTrainingCompleted() {
-        return trainingCourses;
+    public Set<TrainingCourse> getTrainingCompleted() {
+        return Collections.unmodifiableSet(trainingCourses);
```

Comment:
Question: The way that I understand unmodifiable set is that the return set from this method could not be changed but the underlying set could be changed.   In this instance is that what we want or the immutable set where it cannot be changed?

Reply:
Yes to unmodifiable set, unsure as to the change from EnumSet though, not sure how Java handles that stuff under the covers.
I'm happy with the change at the end of the day however.